src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -0,0 +1,680 @@
+// when using the OnOff, Level Control and Color Control as well as the maximal name length of 16 bytes. Putting 256 gives some
+// slack in case different clusters are used. Value obtained by using writer.GetLengthWritten at the end of the SceneTableData
+// Serialize method.
+static constexpr size_t kPersistentSceneBufferMax = 256;
+
+struct SceneTableData : public SceneTableEntry, PersistentData<kPersistentSceneBufferMax>
+{
+    FabricIndex fabric_index = kUndefinedFabricIndex;
+    SceneIndex index         = 0;
+    bool first               = true;
+
+    SceneTableData() : SceneTableEntry() {}
+    SceneTableData(FabricIndex fabric) : fabric_index(fabric) {}
+    SceneTableData(FabricIndex fabric, SceneIndex idx) : fabric_index(fabric), index(idx) {}
+    SceneTableData(FabricIndex fabric, SceneStorageId storageId) : SceneTableEntry(storageId), fabric_index(fabric) {}
+    SceneTableData(FabricIndex fabric, SceneStorageId storageId, SceneData data) :
+        SceneTableEntry(storageId, data), fabric_index(fabric)
+    {}
+
+    CHIP_ERROR UpdateKey(StorageKeyName & key) override
+    {
+        VerifyOrReturnError(kUndefinedFabricIndex != fabric_index, CHIP_ERROR_INVALID_FABRIC_INDEX);
+        key = DefaultStorageKeyAllocator::FabricSceneKey(fabric_index, index);
+        return CHIP_NO_ERROR;
+    }
+
+    void Clear() override { mStorageData.Clear(); }
+
+    CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override
+    {
+        CharSpan NameSpan(mStorageData.mName, mStorageData.mNameLength);
+        TLV::TLVType container;
+        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, container));
+
+        // Scene ID
+        ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneEndpointID), static_cast<uint16_t>(mStorageId.mEndpointId)));
+        ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneGroupID), static_cast<uint16_t>(mStorageId.mGroupId)));
+        ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneID), static_cast<uint8_t>(mStorageId.mSceneId)));
+
+        // Scene Data
+        // A length of 0 means the name wasn't used so it won't get stored
+        if (!NameSpan.empty())
+        {
+            ReturnErrorOnFailure(writer.PutString(TLV::ContextTag(kTagSceneName), NameSpan));
+        }
+
+        ReturnErrorOnFailure(
+            writer.Put(TLV::ContextTag(kTagSceneDTransitionTime), static_cast<uint16_t>(mStorageData.mSceneTransitionTimeSeconds)));
+        ReturnErrorOnFailure(
+            writer.Put(TLV::ContextTag(kTagSceneDTransitionTime100), static_cast<uint8_t>(mStorageData.mTransitionTime100ms)));
+        ReturnErrorOnFailure(mStorageData.mExtensionFieldSets.Serialize(writer));
+
+        return writer.EndContainer(container);
+    }
+
+    CHIP_ERROR Deserialize(TLV::TLVReader & reader) override
+    {
+        CharSpan NameSpan(mStorageData.mName);
```

Comment:
Why would we point this to `mStorageData.mName`?  We never actually want it pointing there....